java/build-jar-multiplatform.sh

```diff
@@ -142,11 +159,15 @@ deploy_jars() {
     fi
   fi
   echo "Start deploying jars"
-  cd "$WORKSPACE_DIR/java"
-  mvn -T16 deploy -Dmaven.test.skip=true -Dcheckstyle.skip -Prelease
-  cd "$WORKSPACE_DIR/streaming/java"
-  mvn -T16 deploy -Dmaven.test.skip=true -Dcheckstyle.skip -Prelease
-  echo "Finished deploying jars"
+  if check_native; then
+    cd "$WORKSPACE_DIR/java"
+    mvn -T16 deploy -Dmaven.test.skip=true -Dcheckstyle.skip -Prelease
+    cd "$WORKSPACE_DIR/streaming/java"
+    mvn -T16 deploy -Dmaven.test.skip=true -Dcheckstyle.skip -Prelease
```

Comment:
It's generally best to scope `cd` to a subshell so that you can be sure you won't be affected if the script continues amidst a signal or error:
```suggestion
    (
      cd "$WORKSPACE_DIR/java"
      mvn -T16 deploy -Dmaven.test.skip=true -Dcheckstyle.skip -Prelease
    )
    (
      cd "$WORKSPACE_DIR/streaming/java"
      mvn -T16 deploy -Dmaven.test.skip=true -Dcheckstyle.skip -Prelease
    )
```

Reply:
Yeah, a subshell would be better